packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearch.tsx

```diff
@@ -276,100 +285,122 @@ function TokensPanel({
     if (!isSuccessful) {
       setErrorMessage(error)
     }
-  }
+  }, [newToken, token])
 
-  const addNewToken: FormEventHandler = async e => {
-    e.preventDefault()
+  const tokenSearchFieldChangeHandler = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
     setErrorMessage('')
+    setNewToken(event.target.value)
+  }
 
-    if (!isAddress(newToken) || isAddingToken) {
-      return
-    }
-
-    setIsAddingToken(true)
-
-    try {
-      await storeNewToken()
-    } catch (ex) {
-      console.log(ex)
-    } finally {
-      setIsAddingToken(false)
+  useEffect(() => {
+    async function addNewToken() {
+      const newTokenLowercased = newToken.toLowerCase()
+      const foundL1AddressInTokenList =
+        typeof bridgeTokens?.[newTokenLowercased] !== 'undefined'
+      const foundL2AddressInTokenList =
+        tokensL2AddressOnly.findIndex(token => token === newTokenLowercased) !==
```

Comment:
Here is what I meant with the user token list, might be safer to have `token.toLowerCase()` here